source/distributed/tria.cc

```diff
@@ -1124,10 +1124,545 @@ namespace
+                       cumulative_sizes_fix.begin());
+
+      // Move every piece of packed data into the consecutive buffer.
+      src_data_fix.reserve(quad_cell_relations.size() *
+                           cumulative_sizes_fix.back());
+      for (auto it_data_cell = packed_data.begin();
+           it_data_cell != packed_data.end();
+           ++it_data_cell)
+        {
+          // Move every fraction of packed data into the buffer
+          // reserved for this particular cell.
+          for (auto it_data_callback = it_data_cell->begin();
+               it_data_callback != it_data_cell->end();
+               ++it_data_callback)
+            std::move(it_data_callback->begin(),
+                      it_data_callback->end(),
+                      std::back_inserter(src_data_fix));
+
+          // Fill the remaining space with invalid entries,
+          // if we only packed the CellStatus information
+          // (i.e. encountered a cell flagged CELL_INVALID).
+          if (it_data_cell->size() == 1)
+            {
+              const size_t bytes_skipped =
+                cumulative_sizes_fix.back() - cumulative_sizes_fix.front();
+
+              src_data_fix.insert(src_data_fix.end(),
+                                  bytes_skipped,
+                                  static_cast<char>(-1)); // invalid_char
+            }
+        }
+
+      // Double check that we packed everything correctly.
+      Assert(src_data_fix.size() == src_data_fix.capacity(),
+             ExcInternalError());
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::DataTransfer::execute_transfer(
+      const typename dealii::internal::p4est::types<dim>::forest
+        *parallel_forest,
+      const typename dealii::internal::p4est::types<dim>::gloidx
+        *previous_global_first_quadrant)
+    {
+      Assert(cumulative_sizes_fix.size() > 0,
+             ExcMessage("No data has been packed!"));
+
+      // Resize memory according to the data that we will receive.
+      dest_data_fix.resize(parallel_forest->local_num_quadrants *
+                           cumulative_sizes_fix.back());
+
+      // Execute fixed size transfer.
+      dealii::internal::p4est::functions<dim>::transfer_fixed(
+        parallel_forest->global_first_quadrant,
+        previous_global_first_quadrant,
+        parallel_forest->mpicomm,
+        0,
+        dest_data_fix.data(),
+        src_data_fix.data(),
+        cumulative_sizes_fix.back());
+
+      // Release memory of previously packed data.
+      src_data_fix.clear();
+      src_data_fix.shrink_to_fit();
+
+      // TODO: for variable transfer
+      // allocate memory for transfer
+      // p4est transfer fixed_begin for transfer of var data sizes
+      // p4est transfer custom for transfer of var data
+    }
+
+
+
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::DataTransfer::unpack_cellstatus(
```

Comment:
this fits the style in this file, but globally we would probably have called this function `unpack_cell_status`. Your choice which way you want to go.